solidity/test/rewards/TestECDSABackportRewards.js

```diff
@@ -0,0 +1,258 @@
+const {accounts, contract, web3} = require("@openzeppelin/test-environment")
+const {createSnapshot, restoreSnapshot} = require("../helpers/snapshot")
+
+const {expectRevert} = require("@openzeppelin/test-helpers")
+
+const KeepToken = contract.fromArtifact("KeepToken")
+const StackLib = contract.fromArtifact("StackLib")
+const KeepRegistry = contract.fromArtifact("KeepRegistry")
+const BondedECDSAKeepFactoryStub = contract.fromArtifact(
+  "BondedECDSAKeepFactoryStub"
+)
+const KeepBonding = contract.fromArtifact("KeepBonding")
+const MinimumStakeSchedule = contract.fromArtifact("MinimumStakeSchedule")
+const GrantStaking = contract.fromArtifact("GrantStaking")
+const Locks = contract.fromArtifact("Locks")
+const TopUps = contract.fromArtifact("TopUps")
+const TokenStakingEscrow = contract.fromArtifact("TokenStakingEscrow")
+const TokenStaking = contract.fromArtifact("TokenStakingStub")
+const TokenGrant = contract.fromArtifact("TokenGrant")
+const BondedSortitionPoolFactory = contract.fromArtifact(
+  "BondedSortitionPoolFactory"
+)
+const RandomBeaconStub = contract.fromArtifact("RandomBeaconStub")
+const BondedECDSAKeep = contract.fromArtifact("BondedECDSAKeep")
+const ECDSABackportRewards = contract.fromArtifact("ECDSABackportRewards")
+
+const KeepTokenGrant = contract.fromArtifact("TokenGrant")
+
+const BN = web3.utils.BN
+
+const chai = require("chai")
+chai.use(require("bn-chai")(BN))
+const expect = chai.expect
+
+describe("ECDSABackportRewards", () => {
+  let registry
+  let rewardsContract
+  let keepToken
+
+  let tokenStaking
+  let tokenGrant
+  let keepFactory
+  let bondedSortitionPoolFactory
+  let keepBonding
+  let randomBeacon
+  let keepMembers
+
+  const owner = accounts[0]
+
+  const keepSize = 16
+  const numberOfCreatedKeeps = 41
+  const keepCreationTimestamp = 1589408353
+
+  // 1,000,000,000 - total KEEP supply
+  //   200,000,000 - 20% of the total supply goes to staker rewards
+  //   180,000,000 - 90% of staker rewards goes to the ECDSA stakers
+  //     1,800,000 - 1% of ECDSA staker rewards goes to May - Sep keeps
+  const ECDSABackportKEEPRewards = 1800000
```

Comment:
To test on real numbers, we need to introduce something like
```
const tokenDecimalMultiplier = web3.utils.toBN(10).pow(web3.utils.toBN(18))
```
and do:
```
const totalRewards = web3.utils.toBN(1800000).mul(tokenDecimalMultiplier)
```

Reply:
Good point, I missed the fact that we also multiply by 10^18 upon deployment. Added.